src/plugins/socketClient.ts

```diff
@@ -150,26 +150,33 @@ export class WebSocketClient {
             // Dispatch with the name of the method, converted to camelCase.
 
             if (d.params && d.params[0]) {
+              const [params, eventtime] = d.params
+
               if (d.method !== 'notify_status_update') {
                 // Normally, we let notifications through with no cache...
-                if (this.store) this.store.dispatch('socket/' + camelCase(d.method), d.params[0])
+                if (this.store) this.store.dispatch('socket/' + camelCase(d.method), params)
               } else {
-                // ...However, status notifications come throug thicc and fast,
+                // ...However, status notifications come through thick and fast,
                 // so we cache these and send them through every second.
-                const date: number = (
-                  d.params[1] === 2 &&
-                  isNaN(d.params[1])
-                )
-                  ? d.params[1]
-                  : new Date().getTime()
+
+                // If any of these properties exist, bypass the cache and send immediately
+                for (const key of ['motion_report']) {
+                  if (this.store && key in params) {
+                    this.store.dispatch('printer/onFastNotifyStatusUpdate', { key, payload: params[key] }, { root: true })
+                    delete params[key]
+                  }
+                }
+
+                const timestamp = eventtime ? eventtime * 1000 : new Date().getTime()
```

Comment:
would be best to use `Date.now()` here, as `new Date().getTime()` is about a [third slower](https://jsbench.me/r7l2hmqupz/1)
```suggestion
                const timestamp = eventtime ? eventtime * 1000 : Date.now()
```

Reply:
I've merged your suggestion, but in all fairness, I doubt the code ever goes there as Moonraker always sends an `eventtime`! 🙂